src/rules/selectorsFormatRule.ts

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) 2019-present, ProductBoard, Inc.
+ * All rights reserved.
+ */
+
+import * as Lint from 'tslint';
+import * as ts from 'typescript';
+
+export class Rule extends Lint.Rules.AbstractRule {
+  public apply(sourceFile: ts.SourceFile): Array<Lint.RuleFailure> {
+    return this.applyWithWalker(
+      new SelectorsFormatRule(sourceFile, this.getOptions()),
+    );
+  }
+}
+
+type Configuration = {
+  reference: string;
+  // TODO: do we wanna make it optional? Does fallback on line `68` make sense?
+  importPaths?: { [key: string]: string | Array<string> };
+};
+
+class SelectorsFormatRule extends Lint.RuleWalker {
+  private configuration: Configuration;
+  private validIdentifiers: Set<string> = new Set();
+
+  constructor(sourceFile: ts.SourceFile, options: Lint.IOptions) {
+    super(sourceFile, options);
+
+    const configuration = {
+      reference: '',
+      ...options.ruleArguments[0],
+    } as Configuration;
+    this.configuration = configuration;
+
+    // TODO: check if configuration is valid in runtime?
```

Comment:
^ yep, let's do something like 

```jsx
if (!configuration.importPaths) throw new Error("...");
```

not a perfect neither but should cover most the cases. Thoughts?